DifferentiationInterface/ext/DifferentiationInterfaceMooncakeExt/onearg.jl

```diff
@@ -45,8 +49,8 @@ function DI.value_and_pullback!(
     contexts::Vararg{Context,C},
 ) where {Y,C}
     dx, dy = only(tx), only(ty)
-    dy_righttype = convert(tangent_type(Y), dy)
-    dx_righttype = set_to_zero!!(convert(tangent_type(typeof(x)), dx))
+    dy_righttype = dy isa tangent_type(Y) ? dy : copyto!!(prep.dy_righttype, dy)
+    dx_righttype = set_to_zero!!(prep.dx_righttype)
```

Comment:
I'm a little confused as to how this function ever gets the right answer. How does `tx` wind up in the `__value_and_pullback!!` call? (which presumably it must do somehow if it's to get the right answer)

Reply:
There's no `dx`, only `dy` for the pullback. And the idea is that I store `dy` in `dy_righttype`, avoiding another allocation if the type is already right

Reply:
Sorry, I'm a little confused -- what's the purpose of the `tx` argument at the top of this method?

Reply:
Ohhh sorry, I didn't see the relationship between `tx` and `dx`. This makes sense. Thanks for clarifying.